backend/services/implementations/courseService.ts

```diff
@@ -36,9 +36,16 @@ class CourseService implements ICourseService {
     };
   }
 
-  async getCourses(): Promise<CourseResponseDTO[]> {
+  async getCourses(onlyPublicCourses: boolean): Promise<CourseResponseDTO[]> {
     try {
-      const courses: Array<Course> = await MgCourse.find();
+      const queryConditions = onlyPublicCourses
+        ? {
+            private: false,
+            published: true,
```

Comment:
Published is a property of modules, not courses. I believe we will have to query the documents based on private status, and the remove the modules that are not published (for staff). Because of this, I think this PR should wait for Shrinjay's module implementation